### ISSUE TITLE
Totara 12: Error when trying to set a Join open date/time

### ISSUE THREAD

**Issue by danmarsden** [blindsidenetworks/moodle-mod_bigbluebuttonbn#217]
use of undefined constant CALENDAR_EVENT_TYPE_ACTION

It seems that although Totara 12 is based on Moodle 3.3, it doesn't include this particular patch which adds this param: https://tracker.moodle.org/browse/MDL-57442

I think the fix is to just comment out this line:
https://github.com/blindsidenetworks/moodle-mod_bigbluebuttonbn/blob/c8683eafa36a53dd14d0698c460e1facf5a73e2f/lib.php#L803

but we'll test and see if there's a moodle friendly way of patching this so that the current branch can support both recent Totara versions and Moodle.

**Comment by jfederico** [comment]
Thanks for bringing that up

**Comment by danvey** [comment]
Any update on this? we are also experiencing the same issue in Totara 12.24.
Anyone know what the impact would be to comment out line 803?

**Comment by danvey** [comment]
Using older version of BBB plugin solves this FYI

**Comment by jfederico** [comment]
@danmarsden is this problem only happening on Totara 12? I tried the latest code on T10, T11, and T13. It worked fine. So 2 things:

1. I have never had code (or access to a server with T12), would you be willing to give me access. That has always been the problem with Totara, we are hand tied to make things compatible because it is closed source.

2. Can you provide the steps to reproduce the problem?

**Comment by danmarsden** [comment]
Hey @jfederico , I haven't looked at it recently sorry - few things on at the moment. We are planning to add it to a T13 site in the next few weeks so I'll see if we can come back to this.

You might just need to rely on external contributions to maintain support for Totara - as they continue to digress further away from core Moodle api's it may become more difficult in future to support anyway!

@alexmorrisnz - do you know if we're still seeing this on T12 without any of our custom patches?

**Comment by jfederico** [comment]
Yeah, I heard from another Totara partner that they had issues as well, but not sure what exactly. That is what brought me back to this long time opened issue.

And you are right, I wish we could keep the code aligned, but it is not possible without access to their source. Only Totara partners may be able to accomplish this.

Thanks

**Comment by alexmorrisnz** [comment]
@danmarsden I haven't heard of anyone getting this error but `CALENDAR_EVENT_TYPE_ACTION` remains undefined on our T12-13's so presumably it is happening
